### PR TITLE
Add menus to game to handle user input

### PR DESCRIPTION
**Pull request by maunika7**

### CODE REVIEW

**Comment by maunika7** [comment]
I changed the spacing from tabs to spaces in editorconfig which is why some of the changes are just spacing.

File: src/app.ts [hunk 1]
```diff
@@ -9,90 +9,149 @@ import Table from './table';
...
+        const tablePosition = this.table.actor.transform.local.position;
+        menu.createMenuBackground(
+            'menu-background', 
+            new MRE.Vector3(1, 0.53, 0.01), 
+            tablePosition.add(new MRE.Vector3(0, 1.3, 0.01)));
+        menu.createMenuText(
+            'menu-text', 
+            'Click Button Below to Start Game', 
+            0.05, 
+            tablePosition.add(new MRE.Vector3(0, 1.5, 0)));
```
Comment (tymlipari) [inline src/app.ts:65]: Could this all be encapsulated into the `Menu` class, and instead the menu constructor takes an array of options with callbacks?

So something like:

```ts
const options = [new MenuText("Click Button Below to Start Game"), new MenuButton("Start", _ => this.onStartGameClicked())];
const menu = new Menu(this.context, options); 
```

Comment (tymlipari) [reply src/app.ts:65]: Then the menu could do some primitive placement logic to build like a stack panel or something out of the options

File: src/menu.ts [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+import Cards from './app';
+
+export default class Menu
+{
+    public parentMenu: MRE.Actor = null;
+
+    constructor(private context: MRE.Context)
+    {
+        // Create a parent object for all the menu items.
+        this.parentMenu = MRE.Actor.Create(this.context, {});
```
Comment (tymlipari) [inline src/menu.ts]: nit: `parentActor` or `containerActor` probably makes more sense

File: src/player.ts [hunk 3]
```diff
@@ -9,102 +9,276 @@ import { TexasHoldEmHand } from '@rgerd/poker-rank';
...
+    constructor(private user: MRE.User, private attachHand = Player.defaultAttachHand) 
+    {
+        this.userId = user.id;
+
+        // Create GroupMask unique to player
+        this.mask = new MRE.GroupMask(Cards.AssetContainer.context, [this.userId.toString()]);
+        user.groups = this.mask;
+    }
+
+    public selectBetAction(currentBet: number)
```
Comment (tymlipari) [inline src/player.ts]: Add the return type (`string`?)

File: src/player.ts [hunk 4]
```diff
@@ -9,102 +9,276 @@ import { TexasHoldEmHand } from '@rgerd/poker-rank';
...
+        while (betDecision === null)
+        {
+            betDecision = this.checkDecision();
+        }
+        
+        this.decision = null;
+        return betDecision;
+    }
+
+    private checkDecision()
```
Comment (tymlipari) [inline src/player.ts]: Does this need to be its own function since it's just returning a private member?

File: src/player.ts [hunk 5]
```diff
@@ -9,102 +9,276 @@ import { TexasHoldEmHand } from '@rgerd/poker-rank';
...
+    {
+        return this.decision;
+    }
+
+    private createPlayerMenu()
```
Comment (tymlipari) [inline src/player.ts:51]: Do we need to recreate the menu after each action? Or can it just be hidden?

**Review by tymlipari — APPROVED** [review]

**Pull request merged**